phase/analyzer.go

```diff
@@ -87,9 +86,6 @@ func (a *Analyzer) Analyze() (files.Analyzed, error) {
 			if err != nil {
 				return files.Analyzed{}, errors.Wrap(err, "unpacking metadata from image")
 			}
-			if atm.OS == "" {
-				platform.GetTargetOSFromFileSystem(&fsutil.Detect{}, atm, a.Logger)
-			}
```

Comment:
Unrelated bug. This only runs if os is blank (which shouldn't happen) but it will also populate distro information and I don't think we can assume this will be the same as the run image as `analyze` often runs in the context of the lifecycle image.